lib/library.cpp

```diff
@@ -730,6 +747,33 @@ bool Library::isargvalid(const Token *ftok, int argnr, const MathLib::bigint arg
     return false;
 }
 
+bool Library::isargvalid(const Token *ftok, int argnr, double argvalue) const
```

Comment:
hmm.. we now have copy/paste I wonder if we can avoid it without using double for the integers. one thing that can be done is creating a utility function that creates the token list.

Reply:
I extracted the token list creation to a separate function.  There's still a lot of copy/paste but I couldn't figure out a good way to remove it.